libcontainer/cgroups/fs/fs.go

```diff
@@ -55,6 +55,23 @@ type manager struct {
 }
 
 func NewManager(cg *configs.Cgroup, paths map[string]string) (cgroups.Manager, error) {
+	// Some v1 controllers (cpu, cpuset, and devices) expect
+	// cgroups.Resources to not be nil in Apply.
```

Comment:
I'm not 100% sure this is the right thing to do. Maybe we should not require it.

It is required because
1. cpu needs to set RT params, see https://github.com/opencontainers/runc/blob/79185bc80666a519867747dac398546da7902fda/libcontainer/cgroups/fs/cpu.go#L26-L29
2. cpuset needs to set cpus and mems in the parent cgroup (yes this is super ugly), see https://github.com/opencontainers/runc/blob/79185bc80666a519867747dac398546da7902fda/libcontainer/cgroups/fs/cpuset.go#L150-L166
3. devices need to check SkipDevices flag, see https://github.com/opencontainers/runc/blob/79185bc80666a519867747dac398546da7902fda/libcontainer/cgroups/fs/devices.go#L24

Now, we can modify all the three places to allow `nil` (and assume RT, cpus/mems, and SkipDevices are not set), and drop this requirement. After all, it is only for some specific cases, and not always required.

Honestly I don't know which way is better.

Reply:
I think that creating a cgroup should be separated from the applying values, so my preference is to allow nil at those 3 places and remove this check.

Reply:
> creating a cgroup should be separated from the applying values

Alas, with cgroup v1 it is not possible in some cases (outlined above), so as much as we want to separate create and set, we can not.

In this case, though, we can allow `nil` (the price to pay is to fail later on Apply in case either of CPU RT priority, cpus/mems, SkipDevices is set).

Reply:
OTOH it's easier to require non-nil Resources here, otherwise it's hard to explain later why cpu/cpuset/devices configuration has unexpectedly failed.

Reply:
Yes, let's leave it as is, unless there are other opinions.

Reply:
I think it's better to leave it as-is for the moment. The fact that we need to configure other cgroups for cpu makes me think that allowing nil Resources will make this even more ugly than it is today.